Makefile

```diff
@@ -1,21 +1,91 @@
-RETOOL=$(CURDIR)/_tools/bin/retool
-PATH := ${PWD}/bin:${PWD}/ENV/bin:${PATH}
+SHELL := /bin/bash -o pipefail
+UNAME_OS := $(shell uname -s)
+UNAME_ARCH := $(shell uname -m)
+
+TMP_BASE := .tmp
```

Comment:
For clarity, I would name this `.tools` and all the variables `.tmp` because (unless I am reading this wrong) the folder isn't really temporary and lasts as long as you work on twirp.

Reply:
It's mean to be temporary, and cleaned easily - I added `make clean` that just `rm -rf .tmp` which is a common thing I do. It's .gitignored, so this works nicely.